src/domain/siwe/siwe.repository.ts

```diff
@@ -1,25 +1,49 @@
-import { ISiweApi } from '@/domain/interfaces/siwe-api.interface';
+import { IConfigurationService } from '@/config/configuration.service.interface';
+import { CacheRouter } from '@/datasources/cache/cache.router';
+import {
+  CacheService,
+  ICacheService,
+} from '@/datasources/cache/cache.service.interface';
 import { ISiweRepository } from '@/domain/siwe/siwe.repository.interface';
-import { VerifyAuthMessageDto } from '@/routes/auth/entities/verify-auth-message.dto.entity';
+import { LoggingService, ILoggingService } from '@/logging/logging.interface';
 import { Inject, Injectable } from '@nestjs/common';
+import { verifyMessage } from 'viem';
+import {
+  generateSiweNonce,
+  parseSiweMessage,
+  validateSiweMessage,
+} from 'viem/siwe';
 
 @Injectable()
 export class SiweRepository implements ISiweRepository {
+  private readonly nonceTtlInSeconds: number;
+  private readonly maxValidityPeriodInSeconds: number;
+
   constructor(
-    @Inject(ISiweApi)
-    private readonly siweApi: ISiweApi,
-  ) {}
+    @Inject(IConfigurationService)
+    private readonly configurationService: IConfigurationService,
+    @Inject(LoggingService)
+    private readonly loggingService: ILoggingService,
+    @Inject(CacheService) private readonly cacheService: ICacheService,
```

Comment:
While I agree with accessing `view/siwe` functions directly from the domain, I also think it's a good idea to maintain the `CacheService/CacheRouter` usage within the datasources layer. Wdyt?

If you agree with that, maybe we could keep `SiweApiService` as a tiny/dummy class containing only the methods to encapsulate the cache calls, by keeping `storeNonce`, `getNonce` and `clearNonce` functions only in its interface.

Reply:
Agreed. I integrated your suggestion in c21988dcce5337f145a7ba04744c2aac15e88d52.